packages/server/src/main/java/software/uncharted/terarium/hmiserver/service/data/DKGService.java

```diff
@@ -37,6 +116,53 @@ public List<DKG> searchClimateDKG(
 		return searchDKG(elasticConfig.getClimateDKGIndex(), page, pageSize, q, source);
 	}
 
+	/**
+	 * Get a DKG entity by curie in the EpiDKG index
+	 * @param curie curie of the entity
+	 * @return list of DKG entities
+	 * @throws IOException
+	 */
+	@Observed(name = "function_profile")
+	public List<DKG> getEpiEntity(final String curie) throws IOException {
+		return getEntity(elasticConfig.getEpiDKGIndex(), curie);
+	}
+
+	/**
+	 * Get a DKG entity by curie in the ClimateDKG index
+	 * @param curie curie of the entity
+	 * @return list of DKG entities
+	 * @throws IOException
+	 */
+	@Observed(name = "function_profile")
+	public List<DKG> getClimateEntity(final String curie) throws IOException {
+		return getEntity(elasticConfig.getClimateDKGIndex(), curie);
+	}
+
+	private SearchResponse<DKG> knnSearchDKG(
+		final String index,
+		final Integer page,
+		final Integer pageSize,
+		final Integer k,
+		final String text,
+		final SourceConfig source
+	) throws IOException, ExecutionException, InterruptedException, TimeoutException {
+		if (k > pageSize) {
+			return null; //error case
+		}
+		KnnQuery knn = null;
+		if (text != null && !text.isEmpty()) {
```

Comment:
Shouldn't you return an error or null if there is no text or it's empty?
What is the point of doing a search with this.